app/actions/websocket/preferences.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import DatabaseManager from '@database/manager';
+import {WebSocketMessage} from '@typings/api/websocket';
+
+export async function handlePreferenceChangedEvent(serverUrl: string, msg: WebSocketMessage): Promise<void> {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+
+    try {
+        const preferences = JSON.parse(msg.data.preferences) as PreferenceType[];
+        const operator = database?.operator;
+        if (operator) {
+            operator.handlePreferences({
+                prepareRecordsOnly: false,
+                preferences,
+            });
+        }
+    } catch (error) {
+        // Do nothing
+    }
+}
+
+export async function handlePreferencesChangedEvent(serverUrl: string, msg: WebSocketMessage): Promise<void> {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+
+    try {
+        const preferences = JSON.parse(msg.data.preferences) as PreferenceType[];
```

Comment:
The `save` post is a user preference and will need to account for the following situation. 
1) user saves a post from another client
* mobile user is viewing saved posts screen
* webapp user saves a post
* mobile should see this new post in the posts list in the saved posts screen
* action is to fetch newly saved post after the websocket received

2) user unsaves a post from another client
* mobile user is viewing saved posts screen
* webapp user unsaves a post
* mobile user should have this post removed from the posts list in the saved posts screen
* Action is to remove the post from the posts lists state in the saved posts screen

Reply:
If I am not mistaken, this should be handled beautifully by the database observables:
- This will change the preferences
- The saved posts screen should be observing the preferences
- The saved posts screen should recreate the list of posts based on the preferences